pynicotine/gtkgui/userbrowse.py

```diff
@@ -688,34 +712,277 @@ def on_select_dir(self, selection):
 
         self.set_directory(iterator.user_data)
 
+    def on_select_file(self, selection):
+
+        if selection is None:
+            return
+
+        self.num_selected_files = selection.count_selected_rows()
+
     def on_folder_key_press_event(self, *args):
 
         keyval, keycode, state, widget = get_key_press_event_args(*args)
+        path, _focus_column = self.FolderTreeView.get_cursor()
+
+        # Note: The "Return" key is handled by on_folder_row_activated
+
+        # Shift bindings need to be above the non-shifted one
+        keycodes, mods = parse_accelerator("<Shift>Right")
+
+        if state & mods and keycode in keycodes:
+            if int(len(self.file_store)) > 0:
+                # Shift focus across to files
+                self.FileTreeView.grab_focus()
+                return True
+
+        # Expand folder node with arrow cursor
         keycodes, mods = parse_accelerator("Right")
 
-        if keycode in keycodes:
-            path, _focus_column = self.FolderTreeView.get_cursor()
+        if keycode in keycodes or (int(keyval) == 61):  # Laptop unshifted + "=" key
```

Comment:
The "+" key on most laptops (that don't have numeric keypads) shares a key with "=".

This keyval solution is rather ugly, but I am unaware of a more graceful way. I haven't tested this for international compatibility yet, but I have a feeling this might cause trouble.

Reply:
Keyvals have caused us trouble on international keyboards previously, so they're indeed a rather ugly solution.

We do something like this in a few other locations: https://github.com/nicotine-plus/nicotine-plus/blob/20bb2450353f63d1cafd0e221388202f62b35905/pynicotine/gtkgui/frame.py#L1578-L1579